### PR TITLE
Add ColorMatrix class and ColorMatrixEffect shader effect

### LINKED ISSUE #1384: Add ColorMatrix ShaderEffect for global color transforms
## Summary

Add a `ColorMatrixEffect` ShaderEffect preset that applies a 4x4 color transformation matrix to a renderable or camera output. A single matrix can express brightness, contrast, saturation, hue rotation, channel swapping, tinting, sepia, inversion, and any combination of these — replacing the need to stack multiple individual effects.

## API Sketch

```js
import { ColorMatrixEffect } from "melonjs";

// per-renderable
sprite.shader = new ColorMatrixEffect(renderer, {
    matrix: [
        1, 0, 0, 0,
        0, 1, 0, 0,
        0, 0, 1, 0,
        0, 0, 0, 1
    ]
});

// on camera (requires #1367 FBO pipeline)
camera.shader = new ColorMatrixEffect(renderer, {
    matrix: ColorMatrixEffect.desaturate(0.5)
});
```

## Built-in presets (static helpers)

- `ColorMatrixEffect.brightness(amount)` — adjust brightness
- `ColorMatrixEffect.contrast(amount)` — adjust contrast
- `ColorMatrixEffect.saturate(amount)` — adjust saturation (0 = grayscale, 1 = normal)
- `ColorMatrixEffect.hueRotate(angle)` — rotate hue in radians
- `ColorMatrixEffect.sepia(amount)` — sepia tone
- `ColorMatrixEffect.invert(amount)` — color inversion
- `ColorMatrixEffect.multiply(matA, matB)` — combine two matrices

## Shader

The fragment shader multiplies each pixel's `vec4(r, g, b, a)` by the 4x4 uniform matrix. Single uniform, single multiply — very cheap.

```glsl
vec4 apply(vec4 color, vec2 uv) {
    return uMatrix * color;
}
```

## Dependencies

- Works on per-renderable today via existing `ShaderEffect` infrastructure
- Camera-level usage requires #1367 (post-processing pipeline with FBO support)

## References

- `ShaderEffect`: `src/video/webgl/shadereffect.js`
- Existing effect presets: `src/video/webgl/effects/`
- CSS `feColorMatrix` spec for matrix layout reference

### PR DESCRIPTION
**Pull request by obiot**
## Summary

Closes #1384 (Phase 1)

### ColorMatrix utility class
- Extends `Matrix3d` with chainable color adjustment methods
- `brightness()`, `contrast()`, `saturate()`, `hueRotate()`, `sepia()`, `invertColors()`
- Zero-allocation: uses `Matrix3d.transform()` internally — no temporary matrices created
- Composable: `new ColorMatrix().brightness(1.3).contrast(1.5).saturate(0.8)`

### ColorMatrixEffect shader effect
- Applies a `ColorMatrix` to any renderable or camera via the existing shader system
- Exposes the same chainable color methods that auto-push the GPU uniform
- `reset()` to clear back to identity
- `multiply()` and `transform()` for advanced use

```js
// simple usage
sprite.shader = new ColorMatrixEffect(renderer).saturate(0.0);

// combined transforms
camera.shader = new ColorMatrixEffect(renderer)
    .brightness(1.3)
    .contrast(1.5)
    .sepia(0.3);

// update dynamically
effect.reset().hueRotate(angle);
```

### Refactored effects
- `SepiaEffect`, `InvertEffect`, `DesaturateEffect` now extend `ColorMatrixEffect`
- Share a single color matrix shader (`mat4 * color`) instead of individual GLSL programs
- `setIntensity()` uses `reset()` + parent color method — no matrix allocation

### Matrix3d.transform()
- New method accepting 16 values (full 4x4) or 6 values (2D affine, promoted to 4x4)
- Mirrors `Matrix2d.transform()` API with aligned JSDoc
- Used by `ColorMatrix` for allocation-free matrix composition

## Test plan

- [x] 23 ColorMatrix tests (all methods, chaining, reset, edge cases)
- [x] 15 Matrix3d.transform() tests (6-arg and 16-arg forms)
- [x] 10 Matrix2d.transform() tests (aligned coverage)
- [x] 2597 total tests pass
- [ ] Existing SepiaEffect/InvertEffect/DesaturateEffect behavior unchanged
- [ ] ColorMatrixEffect works on sprites and cameras

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/melonjs/src/math/matrix3d.ts [hunk 1]
```diff
@@ -251,6 +251,152 @@ export class Matrix3d {
...
+			c7 = b13!;
+			c8 = b20!;
+			c9 = b21!;
+			c10 = b22!;
+			c11 = b23!;
+			c12 = b30!;
+			c13 = b31!;
+			c14 = b32!;
+			c15 = b33!;
+		}
```
[the remark this answers is not in the document]
Comment (obiot) [reply packages/melonjs/src/math/matrix3d.ts]: Fixed in f1ad9dd — transform() now throws an error if arg count is not 6 or 16.

File: packages/melonjs/src/video/webgl/effects/colorMatrix.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	 * @param {number} [n] - value
+	 * @param {number} [o] - value
+	 * @param {number} [p] - value
+	 * @returns {this} this instance for chaining
+	 */
+	transform(...args) {
+		this._matrix.transform(...args);
+		this._syncUniform();
+		return this;
+	}
```
[the remark this answers is not in the document]
Comment (obiot) [reply packages/melonjs/src/video/webgl/effects/colorMatrix.js:149]: Fixed in f1ad9dd — JSDoc now documents both 6-arg (2D affine) and 16-arg (full 4x4) forms.

File: packages/melonjs/src/math/matrix3d.ts [hunk 3]
```diff
@@ -251,6 +251,152 @@ export class Matrix3d {
...
 	}
 
+	/**
+	 * Multiplies the current transformation with the matrix described by the arguments of this method.
+	 * Accepts either 16 values (full 4x4) or 6 values (2D affine: a, b, c, d, tx, ty — promoted to 4x4).
+	 * @returns Reference to this object for method chaining
+	 */
+	/**
+	 * Multiplies the current transformation with a 2D affine matrix.
+	 * The 2D matrix is promoted to 4x4 internally.
```
[the remark this answers is not in the document]
Comment (obiot) [reply packages/melonjs/src/math/matrix3d.ts]: Fixed in f1ad9dd — merged the duplicate JSDoc blocks into a single block on the first overload signature.
